src/Features/Core/Portable/ConvertAutoPropertyToFullProperty/AbstractConvertAutoPropertyToFullPropertyCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,238 @@
+                .AddRange(GetDefaultRule(ImmutableArray.Create<ModifierKind>(), "_"));
+            return rules;
+        }
+
+        private static string GenerateFieldName(
+            IPropertySymbol property, 
+            ImmutableArray<NamingRule> rules)
+        {
+            var propertyName = property.Name;
+            var fieldName = "";
+            var isStatic = property.IsStatic;
+            foreach (var rule in rules)
+            {
+                if (rule.SymbolSpecification.AppliesTo(
+                    new SymbolKindOrTypeKind(SymbolKind.Field), 
+                    isStatic ? DeclarationModifiers.Static : DeclarationModifiers.None, 
+                    Accessibility.Private))
+                {
+                    fieldName = rule.NamingStyle.MakeCompliant(propertyName).Single();
+                    break;
+                }
+            }
+
+            var uniqueName = NameGenerator.GenerateUniqueName(
+                fieldName, n => property.ContainingType.GetMembers(n).IsEmpty);
+            return uniqueName;
+        }
+
+        private async Task<SyntaxNode> ExpandPropertyAndAddFieldAsync(
+            Document document, 
+            SyntaxNode property, 
+            SyntaxNode getAccessor, 
+            SyntaxNode setAccessor, 
+            SyntaxNode root, 
+            IPropertySymbol propertySymbol, 
+            string fieldName, 
+            SyntaxGenerator generator, 
+            CancellationToken cancellationToken)
+        {
+            var workspace = document.Project.Solution.Workspace;
+
+            // add statements to existing accessors
+            var getAccessorStatements = new SyntaxNode[] {
+                generator.ReturnStatement(
+                    generator.IdentifierName(fieldName)) };
+            var newGetAccessor = await AddStatementsToAccessorAsync(
+                document, 
+                getAccessor, 
+                getAccessorStatements, 
+                generator, 
+                cancellationToken).ConfigureAwait(false);
+
+            var setAccessorStatements = new SyntaxNode[] {
+                generator.ExpressionStatement(generator.AssignmentStatement(
+                    generator.IdentifierName(fieldName),
+                    generator.IdentifierName("value"))) };
+            var newSetAccessor = await AddStatementsToAccessorAsync(
+                document, 
+                setAccessor, 
+                setAccessorStatements, 
+                generator, 
+                cancellationToken).ConfigureAwait(false);
+
+            var newProperty = generator
+                .WithAccessorDeclarations(property, new SyntaxNode[] { newGetAccessor, newSetAccessor})
+                .WithAdditionalAnnotations(Formatter.Annotation)
+                .WithAdditionalAnnotations(new SyntaxAnnotation("property"));
+            newProperty = await Formatter.FormatAsync(newProperty, workspace).ConfigureAwait(false);
```

Comment:
Why do you have to do this here?  Shouldn't it happen automatically when the refactoring is applied at the end?